nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/PutAzureBlobStorage_v12.java

```diff
@@ -121,18 +146,47 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
             if (createContainer && !containerClient.exists()) {
                 containerClient.create();
             }
-            BlobClient blobClient = containerClient.getBlobClient(blobName);
 
+            BlobClient blobClient = containerClient.getBlobClient(blobName);
+            final BlobRequestConditions blobRequestConditions = new BlobRequestConditions();
+            Map<String, String> attributes = new HashMap<>();
```

Comment:
@nandorsoma the attributes are not applied in case of an error; the code will not reach the `session.putAllAttributes(flowFile, attributes)` in this case.

According to the [documentation](https://learn.microsoft.com/en-us/dotnet/api/azure.storage.blobs.models.blobproperties.contenttype?view=azure-dotnet#definition), the default mimetype is "application/octet-stream".

It would seem that `ATTR_NAME_LANG` should be there exactly because of backwards compatibility. It is expected that if the upload succeeds, then that attribute is there and by default, equal to null.

In case of an error, the attribute will not be set (similar to above).

I'll take a look about cleaning up the code to make it less verbose around the important bits.

Reply:
Sorry I wasn't clear. It's my bad. I meant when execution jumps to the catch block and recovers in case of `ignore` resolution. But meanwhile, I realized since `ignore` is not the default behavior, it won't break anything.

Nevertheless, I still have issues with the two properties I mentioned previously. Applying `mime.type` and `lang` before the upload is not ok because it can fail, and then these values don't make sense. I think it's better to apply them after upload, and then we can get the values from the client, which is more precise.

The test you mentioned was written when ignore wasn't in the game; that's why it is probably failing. But I think we can modify it accordingly. Which test is that?

Also, applying the other three attributes before uploading could have a purpose. Still, if we don't have a use case for that, then I think it would be clearer to apply all attributes after a successful upload, like in the original version.

Reply:
Sorry I wasn't clear. It's my bad. I meant when execution jumps to the catch block and recovers in case of `ignore` resolution. But meanwhile, I realized since `ignore` is not the default behavior, it won't break anything.

Nevertheless, I still have issues with the two properties I mentioned previously. Applying `mime.type` and `lang` before the upload is not ok because it can fail, and then these values don't make sense. I think it's better to apply them after upload, and then we can get the values from the client, which is more precise.

The test you mentioned was written when ignore wasn't in the game; that's why it is probably failing. But I think we can modify it accordingly. Which test is that?

Also, applying the other three attributes before uploading could have a purpose. Still, if we don't have a use case for that, then I think it would be clearer to apply all attributes after a successful upload, like in the original version.

Reply:
I see what you mean now – you're right, in the case where we're not replacing the blob, it makes no sense to set those attributes. Fixed in 438c4518b317b71c24fa2633e591fbdd6ecc35e1.